test/common/schemaform/helpers.unit.spec.js

```diff
@@ -285,6 +285,32 @@ describe('Schemaform helpers:', () => {
       expect(newSchema.properties.field['ui:hidden']).to.be.true;
       expect(newSchema).not.to.equal(schema);
     });
+    it.only('should set hidden on nested hidden object field', () => {
+      const schema = {
+        type: 'object',
+        properties: {
+          unhide: { type: 'boolean' },
+          nestedObject: {}
+        }
+      };
+      const uiSchema = {
+        'ui:options': { hideIf: (fieldData) => !fieldData.unhide },
+        nestedObject: {
+          'ui:options': { hideIf: () => true }
+        }
+      };
+      const data = { unhide: false };
+
+      const newSchema = setHiddenFields(schema, uiSchema, data);
+
+      expect(newSchema['ui:hidden']).to.be.true;
+
+      // Unhide the parent object
+      const unhiddenSchema = setHiddenFields(schema, uiSchema, { unhide: true });
```

Comment:
Hmm, I guess it wasn't that it was being unhidden that caused the nested `hideIf` to not be run, but that it had to get past the parent's `hideIf`...Alright. Updating!